### PR TITLE
Improved mock documentation

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

### CODE REVIEW

File: docs/mockwbemserver.rst [hunk 1]
```diff
@@ -10,26 +10,26 @@ Mock WBEM server
 Overview
 --------
 
-The pywbem package contains the ``pywbem_mock`` subpackage which provides mock
-WBEM server support that enables using the pywbem library without a WBEM
-server. This subpackage is useful for testing the pywbem library itself as well
+The 'pywbem_mock' module of pywbem provides mock WBEM server support that
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:13]: Change "mock WBEM server support" to "a mock WBEM server"

File: docs/mockwbemserver.rst [hunk 2]
```diff
@@ -851,50 +853,48 @@ that somewhere an instance the class must be created each time a namespace is
...
-    instance write         "instance-write"    CreateInstance           :ref:`Instance Write Provider`
-    instance write         "instance-write"    ModifyInstance           :ref:`Instance Write Provider`
-    instance write         "instance-write"    DeleteInstance           :ref:`Instance Write Provider`
-    ====================== =================== ======================== ================================
-
-Each user-defined provider is a Python class which is a subclass
-of the corresponding default provider for the provider type as defined in
-the previous table.
+    ============== ================ ============== ==============================
+    Provider type  Type name        WBEM operation Details
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:879]: I prefer more than "Details". However, we do have an issue in that we don't have one clear name for the class that is the defining superclass for user-defined providers.   I used "Default provider class". Works for instances but not really for methods.  Something like "API defining class" but that is awkward. How about "Defining class"?

File: docs/mockwbemserver.rst [hunk 3]
```diff
@@ -851,50 +853,48 @@ that somewhere an instance the class must be created each time a namespace is
...
 Creating user-defined instance providers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
- User-defined instance providers can override the default implementations for
- the methods defined in the above table (``CreateInstance``, ``ModifyInstance``,
- and ``DeleteInstance``). These methods are defined in a user-defined class
- that subclasses :class:`pywbem_mock.InstanceWriteProvider`. The details of
- defining this provider are in section :ref:`Instance Write Provider`.
+
+User-defined instance write providers can override the default implementations for
+the methods defined in the above table (``CreateInstance``, ``ModifyInstance``,
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:894]: Shouldn't the work methods really be "operations"?

File: docs/mockwbemserver.rst [hunk 4]
```diff
@@ -851,50 +853,48 @@ that somewhere an instance the class must be created each time a namespace is
...
+and ``DeleteInstance``). These methods are defined in a user-defined class
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:895]: Change "These methods" to "The user-defined implementation of these methods"

File: docs/mockwbemserver.rst (hunk 1 above)
Comment (andy-maier) [reply docs/mockwbemserver.rst:13]: DONE

File: docs/mockwbemserver.rst (hunk 2 above)
Comment (andy-maier) [reply docs/mockwbemserver.rst:879]: The links in that column target the section in the documentation that describes the provider type, so I changed the column heading to "Description".

In these sections, the language also needs to be cleaned up somewhat. Will do that in the next update.

File: docs/mockwbemserver.rst (hunk 3 above)
Comment (andy-maier) [reply docs/mockwbemserver.rst:894]: Yes. DONE

File: docs/mockwbemserver.rst (hunk 4 above)
Comment (andy-maier) [reply docs/mockwbemserver.rst:895]: Changed the text along these lines, but slightly different: 
```
User-defined instance write providers can override the default implementations
for the corresponding WBEM operations listed in the above table (i.e.
``CreateInstance``, ``ModifyInstance``, and ``DeleteInstance``). The Python
methods implementing these WBEM operations are defined in a user-defined class
that subclasses the :class:`~pywbem_mock.InstanceWriteProvider` class. The
details of implementing this provider type are described in
:ref:`Instance Write Provider`.
```

**Pull request merged**